src/backend/joanie/core/models/products.py

```diff
@@ -660,6 +661,15 @@ def clean(self):
                         "Orders can't be placed on enrollments originating from an order."
                     )
                 )
+            # For attempts of creating an order with a course run that has ended.
+            if self.enrollment.course_run and str(
+                self.enrollment.course_run.course.state
+            ) is str(CourseState.STATE_TEXTS[CourseState.ARCHIVED_CLOSED]):
```

Comment:
It's proper to use state priority instead.

```suggestion
            if self.enrollment.course_run.course.state["priority"] >= CourseState.ARCHIVED_CLOSED:
```

I'm using `>=`, in this way, we are ensuring the course run cannot be `Archived closed` or `To be scheduled`.

Reply:
I totally agree, I've just updated the code.